apps/emqx/src/emqx_channel.erl

```diff
@@ -217,6 +220,9 @@ info(impl, #channel{session = Session}) ->
 set_conn_state(ConnState, Channel) ->
     Channel#channel{conn_state = ConnState}.
 
+set_conninfo_proto_ver({none, #{version := ProtoVer}}, Channel = #channel{conninfo = ConnInfo}) ->
```

Comment:
this is breaking the opaqueness of emqx_frame:options type.
maybe export an API to get protocol:

emqx_frame:get_proto_ver_from_options/1